### ISSUE TITLE
Error in Event when unpickling

### ISSUE THREAD

**Issue by klanderson** [NeuralEnsemble/python-neo#262]
I'm getting an error when trying to load a Neo object that has been saved as a pickle file. It seems like something is not being saved correctly from the the `times` attribute of an `Event`. 

First I load an nev file (Blackrock format)

```
# Load .nev file
data_block = neo.io.BlackrockIO(file_name, verbose=False).read_block(name=file_name,
                                                                   nsx_to_load=None,
                                                                   channels='all',
                                                                   units='all',
                                                                   load_waveforms=False,
                                                                   load_events=True,
                                                                   lazy=False,
                                                                   cascade=True)
```

The events seem to be loaded correctly:

`data_block.list_children_by_class(neo.Event)[0]`

```
Event
name: 'digital_input_port'
description: 'Events of the digital input port'
annotations: {'none': 'none'}
```

`data_block.list_children_by_class(neo.Event)[0].times.units.dimensionality
`
`Dimensionality({second: 1})`

And pickling gives no errors:

```
# Save as pickle
with open(file_name+'.pickle', 'wb') as handle:
  pickle.dump(data_block, handle)
```

But I get an error when loading:

```
# Load pickle
with open(file_name+'.pickle', 'rb') as handle:
  data_block_pickle = pickle.load(handle)
```

Here is the bottom of the error trace:

```
/Users/..../lib/python2.7/site-packages/quantities-0.11.1-py2.7.egg/quantities/quantity.pyc in _reconstruct_quantity(subtype, baseclass, baseshape, basetype)
    738     """
    739     _data = np.ndarray.__new__(baseclass, baseshape, basetype)
--> 740     return subtype.__new__(subtype, _data, dtype=basetype,)

/Users/..../lib/python2.7/site-packages/neo-0.5.0a1-py2.7.egg/neo/core/event.pyc in __new__(cls, times, labels, units, name, description, file_origin, **annotations)
     71                 dim = units.dimensionality
     72             except AttributeError:
---> 73                 raise ValueError('you must specify units')
     74         else:
     75             if hasattr(units, 'dimensionality'):

ValueError: you must specify units

```

Any insight into this error or how to fix? My Neo `Block` only has one `Event` object and it seems to be intact so something is going awry when pickling.

**Comment by grg2rsr** [comment]
not a fix, but a possible workaround (as I had a similar problem), using `dill` instead of `pickle` solved it for me

**Comment by ckemere** [comment]
Unfortunately, Dill gave me the same error (though Python 3.6).

**Comment by apdavison** [comment]
Should be fixed in #291. @klanderson, @grg2rsr and @ckemere : can you try with the latest master? (The problem seems to be in unpickling, so previously-pickled Events should work).
